environment.yml

```diff
@@ -28,6 +28,7 @@ dependencies:
   - python-dateutil>=2.1
   - setuptools_scm
   - wxpython
+  - typing-extensions>=4.0.0
```

Comment:
We already use `typing-extensions` for `Self`; this addition is not needed as a runtime requirement. It is pulled in by mypy.

Reply:
There are other type checkers than `mypy`, for instance `pyright` does not have any python dependencies as it is written in TS. Type-checkers may also run in external virtual environments, such as pre-commit hooks.

IMO if someone created a bare venv and only pip installed matplotlib, then stuff should still type check.

Reply:
I will double-check if it is needed tomorrow.

Reply:
This file is not for venvs, but conda; we don't have any typing-related dependencies here. They're handled by pre-commit or the tox environment.

Reply:
OK, so I tested it: downloaded current release from PyPI and installed in a bare virtualenv together with pyright (python wrapper)

```
$ pip list
Package         Version
--------------- -----------
contourpy       1.2.1
cycler          0.12.1
fonttools       4.53.1
kiwisolver      1.4.5
matplotlib      3.9.1
nodeenv         1.9.1
numpy           2.0.1
packaging       24.1
pillow          10.4.0
pip             24.2
pyparsing       3.1.2
pyright         1.1.373
python-dateutil 2.9.0.post0
setuptools      68.2.2
six             1.16.0
wheel           0.41.2
```

Seems like it works after doing the minimal edit, despite `typing-extensions` not being present in the virtual environment.